### PR TITLE
Research Examines!

### PR DESCRIPTION
**Pull request by GunHog**
Science Goggles and the Prototype Hardsuit Helmet may now scan items for
their research potential and machines for their contents.

Forum thread: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=4823

Examine an item to gain additional data.
- Tech levels
- Materials that can be salvaged from it if recycled
- Reliability rating
- Crit failure, if present.

Examine a machine to learn the components used to build it.

<b>Additional items with the ability may be added. Certain silicons/drones/swarmers may also be considered. 
EDIT: Borgs and Drones added, also now an action button!
EDIT 2: Tech names fixed!</b>

screenshot 2015-10-07 13 39 18
<b> Machine Scanning </b>
screenshot 2015-10-07 16 11 07

<b>Known issues:</b>
- ~~Removing the RD hardsuit helm or Science Goggles while wearing the other causes the scanner var to be removed when it should not have been.~~
  ~~Tech names are not consistent with those given in the Protolathe.~~

### CODE REVIEW

File: code/game/machinery/machinery.dm [hunk 1]
```diff
@@ -406,14 +404,22 @@ Class Procs:
...
+			display_parts(user)
 		if(shouldplaysound)
 			W.play_rped_sound()
 		return 1
 	return 0
 
+/obj/machinery/proc/display_parts(mob/user)
+	user << "<span class='notice'>Following parts detected in the machine:</span>"
+	for(var/var/obj/item/C in component_parts)
+		user << "<span class='notice'>   [C.name]</span>"
```
Comment (RemieRichards) [inline code/game/machinery/machinery.dm]: ```\icon[C] [C.name]``` so they get a nice little picture too.

**Comment by AlexCrimson** [comment]
Id say remove the tech numbers and just have the research type. Maybe give the more detailed examine to RD suit only.

**Comment by GunHog** [comment]
I will have to make Science Goggles printable eventually. But, why remove the levels? That is the information someone wants most when looking for things to DA. As for the RD's hardsuit having the full examine only, I cannot justify that unless I manage to get it added to the primary map.

**Comment by Durkel** [comment]
Cool idea, Though I think we should just remove durability of items. Annoying having to decon eight+ bluespace cells so the damn thing don't break in a mech.

**Comment by GunHog** [comment]
I am not too beg a fan of it either, but so long as it exists, I will have support for it. Only the AE gun and cells really use it.

**Comment by WJohn** [comment]
More player feedback is always nice :+1:

**Comment by Bawhoppen** [comment]
Don't give it to the prototype suit, I don't think that fits IMO, you should need to wear specific goggles if you want to see.

**Comment by Xhuis** [comment]
Seems pretty good. You might want to have an on/off button for this, though, since I could see it getting annoying as chemist getting the glass count of a beaker whenever you examine it to see its reagents.

**Comment by GunHog** [comment]
I have taken your feedback into account, and made it into an action button! It has the plus of having fixed the multiple scanners bug.

I have also given it to Cyborgs and Drones.

screenshot 2015-10-08 11 32 08

**Comment by GunHog** [comment]
Changelog added!

**Comment by Xhuis** [comment]
I'm not sure @GunHog wanted this merged...

**Comment by GunHog** [comment]
I did, that is why I removed the WIP tag :)  I did add your feedback, Xhuis. The only change I would have had left to make is to get a better sprite, or add more functions/objects. Nothing required to make the feature work.

If you have additional questions or concerns, let me know and I can make any needed changes in a new PR.

**Pull request merged**